src/order/ideal.lean

```diff
@@ -135,8 +167,38 @@ begin
   { exact λ _, I.mem_of_le le_top topmem }
 end
 
-lemma proper_of_ne_top {I : ideal P} (ntop : I ≠ ⊤) : proper I :=
-proper_of_not_mem (λ h, ntop (top_of_mem_top h))
+lemma is_proper_of_ne_top {I : ideal P} (ne_top : I ≠ ⊤) : is_proper I :=
+is_proper_of_not_mem (λ h, ne_top (top_of_mem_top h))
+
+lemma is_proper.ne_top {I : ideal P} (hI : is_proper I) : I ≠ ⊤ :=
+begin
+  intro h,
+  rw [ext'_iff, top_coe] at h,
+  apply hI.ne_univ,
+  assumption,
+end
+
+lemma is_proper.of_is_coatom {I : ideal P} (hI : is_coatom I) : is_proper I :=
+is_proper_of_ne_top hI.1
+
+lemma is_proper_iff_ne_top {I : ideal P} : is_proper I ↔ I ≠ ⊤ :=
+⟨λ h, h.ne_top, λ h, is_proper_of_ne_top h⟩
+
+lemma is_maximal.is_coatom {I : ideal P} (h : is_maximal I) : is_coatom I :=
+⟨is_maximal.to_is_proper.ne_top, λ J _, by { rw [ext'_iff, top_coe],
+exact is_maximal.maximal_proper J ‹_› }⟩
+
+lemma is_maximal.is_coatom' {I : ideal P} [is_maximal I] : is_coatom I :=
+is_maximal.is_coatom ‹_›
+
+lemma is_maximal.of_is_coatom {I : ideal P} (hI : is_coatom I) : is_maximal I :=
+{
+  maximal_proper := λ _ _, by simp [hI.2 _ ‹_›],
+  ..is_proper.of_is_coatom ‹_›
+}
```

Comment:
According to [the style guide](https://leanprover-community.github.io/contribute/style.html#structuring-definitions-and-theorems):
```suggestion
{ maximal_proper := λ _ _, by simp [hI.2 _ ‹_›],
  ..is_proper.of_is_coatom ‹_› }
```